### ISSUE TITLE
Support Request: H610PRO not detected, tried cleanup in FAQ

### ISSUE THREAD

**Issue by Coolykoen** [OpenTabletDriver/OpenTabletDriver#1622]
# Support Request
Tablet not detected, no other driver currently installed.

## What I've already tried
I did try the cleanup tool in the FAQ but it still won't detect the tablet.

Then i looked at other issues with detection and saw "tablet > device string reader" but that gives me an error:
![image](https://user-images.githubusercontent.com/48558949/141174567-cb599214-b7b2-4960-859e-9bd89e09c2b0.png)
Yes, i did try to do what it says, but it keeps asking that everytime i replug > press okay > same question in a loop

**Comment by jamesbt365** [comment]
Help > export diagnosis instead of dumping the console, upload it as a txt file please.

**Comment by Coolykoen** [comment]
Maybe i found the issue, will try changing the profile to match the detected hardware:
![image](https://user-images.githubusercontent.com/48558949/141175700-b553a254-47a1-4f4c-b7ad-12b297fe1167.png)

**Comment by Coolykoen** [comment]
nvm thats not it (im dumb, scrolled a bit further down the profile and it was already in there)

**Comment by Coolykoen** [comment]
Here is the dump:
[dump.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7515259/dump.txt)

**Comment by jamesbt365** [comment]
can you: right click on start menu > device manager > view > device by container > look for your tablet there > open the container (arrow on the left) > screenshot device manager

**Comment by Coolykoen** [comment]
Sure :)
![image](https://user-images.githubusercontent.com/48558949/141176563-64fee402-e0a5-40e4-b36e-287ed30dab6f.png)

**Comment by jamesbt365** [comment]
Tablet > device string reader:
9580 in the top box 110 in the middle and click dump all, save that and send it here

**Comment by Coolykoen** [comment]
Here you go, also thanks for your help ^^
[dump2.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7515338/dump2.txt)

**Comment by Mrcubix** [comment]
this tablet require the use of WinUSB and pre-release build to work, but first, do these steps:

close otd, get the oem driver, install it and run it then following the next series of steps:

- https://desowin.org/usbpcap/thankyou.html?file=1.5.4.0/USBPcapSetup-1.5.4.0.exe Download and install this tool

- When done, head to C:\Program Files\USBPcap in the explorer (you can probably copy paste that path in the path bar)

- Once you're there, start USBPcapCMD.exe

- A console will appear, answer n at the first prompt and reboot your pc

- Head back to C:\Program Files\USBPcap,

- Start USBPcapCMD.exe,

- screenshot the console.

**Comment by Mrcubix** [comment]
na, we are just getting the data necessary to know which interface you'll have to install WinUSB on and the config you'll have to use

**Comment by Coolykoen** [comment]
i see, alright sounds good :)

**Comment by Coolykoen** [comment]
Hey, as promised im back, and uh, the usbpcap console didnt give any prompt so i couldnt do this step:
"A console will appear, answer n at the first prompt and reboot your pc"
But here is the console screenshot, with the oem driver installed and running in the background.
![image](https://user-images.githubusercontent.com/48558949/141681600-4e72b89e-2fec-4b25-8625-29be788f818b.png)

**Comment by Mrcubix** [comment]
- type 3 then enter,
- name the file. Press enter and do the following steps:

- Move for 5s
- Unplug the tablet from your computer (remove the usb cable from the port),
- Wait 2 seconds,
- Plug in your tablet again (connect your tablet to your pc ),
- Wait 2 seconds,
- Move for 5s
- Close the console (red cross),
- A new file has been generated with the name you gave in the same folder,
- Upload that file here

**Comment by Coolykoen** [comment]
[tablet.pcap.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7533732/tablet.pcap.txt)
Uploaded as .txt again because .pcap is not allowed :D

**Comment by Mrcubix** [comment]
- Download [Zadig](https://github.com/pbatard/libwdi/releases/download/b755/zadig-2.6.exe)
- Start Zadig
- Options > show all devices
- Screenshot the list of devices

**Comment by Coolykoen** [comment]
![image](https://user-images.githubusercontent.com/48558949/141682326-898b5322-df47-4b05-a923-024e34afce96.png)

**Comment by Mrcubix** [comment]
Install WinUSB (driver on the right side) to `PenTablet (Interface 1)`

**Comment by Coolykoen** [comment]
So after doing that, it said driver installation successful, but opentablet driver is not detecting it yet? 
should i restart both? maybe restart the pc?

![image](https://user-images.githubusercontent.com/48558949/141683119-24236b1b-9e31-416d-bef9-b4be0ff10505.png)
![image](https://user-images.githubusercontent.com/48558949/141683125-0689f530-7f6e-4e88-af24-bb4f6910ca89.png)

**Comment by Mrcubix** [comment]
export a diag now, a new descriptor should have appeared

**Comment by Coolykoen** [comment]
in which program? do i do the usbpcap thing again?

**Comment by Mrcubix** [comment]
In OTD obviously

**Comment by Coolykoen** [comment]
Both of them here:

[new.json.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7533838/new.json.txt)
[new.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7533839/new.txt)

**Comment by Mrcubix** [comment]
https://cdn.discordapp.com/attachments/615939905566015493/890625738691211334/OpenTabletDriver.win-x64.zip you'll need to get this build to continue,
- Close the previous release of OTD
- Extract the build to it's own folder,
- Create a folder named `userdata` at the same location as the exes
- Start the UX
- Export another diag there.

**Comment by Coolykoen** [comment]
So, i cant do one of the two diags cause of this:
![image](https://user-images.githubusercontent.com/48558949/141684229-460dddc7-4aee-4957-b06a-a550be7f47dc.png)
(yes i did replug, got an error that otd deamon crashes, i clicked yes to restart the deamon, same but still got the same error when trying again)

here's the other dia
[newnew.json.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7533854/newnew.json.txt)
g:

**Comment by Mrcubix** [comment]
i asked for a diag, not a string dump, so let me check that 2s

**Comment by Mrcubix** [comment]
i don't see the tablet in the diagnostic file, are you sure it's plugged in?

**Comment by Coolykoen** [comment]
yup :D

**Comment by Coolykoen** [comment]
since the winusb driver was installed, my mouse doesnt move when i use the tablet so maybe that didnt work out somehow?

**Comment by Coolykoen** [comment]
Ill restart my pc, and do a diag again to make sure

**Comment by Mrcubix** [comment]
just replug

**Comment by Coolykoen** [comment]
[newnewnew.json.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7533882/newnewnew.json.txt)

**Comment by Mrcubix** [comment]
[H610 Pro.zip](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7533885/H610.Pro.zip)
extract this in `Configurations/Huion` then either go to `Tablets > Detect tablet` or Restart OTD, you tablet should be detected and your cursor should be moving

**Comment by Coolykoen** [comment]
In the normal OTD release? or the custom version you sent?

**Comment by Mrcubix** [comment]
Pre-release, the latest build i sent

**Comment by Coolykoen** [comment]
In the OTD version you sent, it detect the tablet now (progress :D) however, the cursor wont move.

**Comment by Mrcubix** [comment]
go to `Tablet > tablet debugger` move your pen around, does it report any kind of data?

**Comment by Coolykoen** [comment]
nope, this windows stays empty
![image](https://user-images.githubusercontent.com/48558949/141684949-02db05fb-3844-44f6-9ab6-3435e0ab5ad4.png)

**Comment by Mrcubix** [comment]
export a diag again and upload it here

**Comment by Coolykoen** [comment]
[newnewnewnew.json.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7533912/newnewnewnew.json.txt)

**Comment by Mrcubix** [comment]
wait, did you install WinUSB on all 3 interfaces without my knowledge?

**Comment by Coolykoen** [comment]
i didnt though
interface 2 has the original driver
int 0 seems to have a different version of winusb (even though i used the cleaning batch file before starting)

So no idea how that got there
![image](https://user-images.githubusercontent.com/48558949/141685534-17057205-a4ed-4825-9052-66fca98b40ec.png)
![image](https://user-images.githubusercontent.com/48558949/141685540-7867ddd2-3cfc-4002-badd-197261c66822.png)
![image](https://user-images.githubusercontent.com/48558949/141685542-609b13b5-5119-46d3-aab5-a6ece2dc9606.png)

**Comment by Coolykoen** [comment]
Will do, if you dont mind, another time. probably tomorrow. Thanks for your help so far

**Comment by Coolykoen** [comment]
![image](https://user-images.githubusercontent.com/48558949/141825471-8450d6e5-3eeb-4a1b-9d0e-15b75510063b.png)

**Comment by Mrcubix** [comment]
![image](https://user-images.githubusercontent.com/39861216/141830637-0a579961-9e9b-4572-8972-96fce3fa5210.png)
uninstall both of those, tick the box at the last step to remove the driver, then replug, and install WinUSB on `Pentablet interface 1` using zadig

**Comment by Coolykoen** [comment]
just to make sure we're on the same page, here's the before:
![image](https://user-images.githubusercontent.com/48558949/141831384-a27a7ffc-926c-4fcd-8d6d-322da0751e39.png)

Now i'll install winusb to int 1:
![image](https://user-images.githubusercontent.com/48558949/141831429-78945496-61af-4ce1-907a-df305d1552ad.png)
![image](https://user-images.githubusercontent.com/48558949/141831537-0ce2867b-6ae0-46ba-9e52-2a79244ffaf0.png)
![image](https://user-images.githubusercontent.com/48558949/141831626-19f4310e-5c4c-45a8-a46f-9f240ed6640a.png)


And here's what that looks like after installing:
![image](https://user-images.githubusercontent.com/48558949/141831771-abaa3f3b-262d-4184-9d4f-ee8425b0b70e.png)

Yea it looks better now, lets try OTD again...
Okay, more progress...

this time, i ...kind of... have a working pen.
It doesnt move the mouse almost at all, but sometimes when just tapping and moving quickly, i do see the cursor jump around a bit here and there.

**Comment by Mrcubix** [comment]
export a diagnostic file now from pre-release

**Comment by Coolykoen** [comment]
[newnewnewnewnew.json.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7540830/newnewnewnewnew.json.txt)

**Comment by Mrcubix** [comment]
you indeed installed WinUSB on another interface before, anyway, use this config  
[H610 Pro 2048.zip](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7540841/H610.Pro.2048.zip)
anyway, delete the old config and move this to `Configurations\Huion`

**Comment by Coolykoen** [comment]
YES that did it!
it moves, and i think it moves perfectly at that.

however, the cursor only moves when dragging, not when hovering, so the mouse is always holding left click.
Hmm, any idea whats causing that?

**Comment by Mrcubix** [comment]
lemme check the parser, i'll edit that message when done

**Comment by Coolykoen** [comment]
Im dragging the cursor around on that debugger window, but nothing... just always looks like this:
![image](https://user-images.githubusercontent.com/48558949/141837630-144fa5d1-ef9e-4382-bcef-019528a90fef.png)
I tried turning on "enable data recording" but still nothing

**Comment by Mrcubix** [comment]
Which cursor move, windows ink (small dot or diamond) or windows cursor

**Comment by Coolykoen** [comment]
the small dot one
also it works when otd is not even running... so windows is probably still screwing with it
[newnewnewnewnewnew.json.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7541037/newnewnewnewnewnew.json.txt)

**Comment by Coolykoen** [comment]
![image](https://user-images.githubusercontent.com/48558949/141840345-65576016-56ca-4ae4-be7e-76f69b5f3654.png)
This is looking really good now

the tablet raw data is like:
C0 is for hover, it changes to C1 when dragging

The four next ones are movement, and the last two are for pressure


the cursor now doesnt move at all (neither dot nor mouse)
But this data is very useful i imagine

**Comment by Mrcubix** [comment]
So first of all, Huion is initializing their tablet rong somehow, second, what's the second byte when you press aux keys (tablet keys, not the pen)

**Comment by Coolykoen** [comment]
Well to be fair we are using a config file for the 2048 version whereas this is the H610PRO(8192)
anyway i have to go now, will continue next time, sorry for all the delays, got busy work days :)
Thanks again

**Comment by Mrcubix** [comment]
funny thing is that, i already know what's the max pressure this thing will throw and it's def 2048

**Comment by Coolykoen** [comment]
Question, could we possibly do this in a voice call? screenshare for example would help make things go very quickly
Then we just document here what the issues were and what we did to solve them.

because i'd like to fix it, but spending a week for it seems a bit crazy just to be able to use it haha.

if not, we can ofc continue here though.

**Comment by Mrcubix** [comment]
we have a [discord server](https://discord.gg/P3GYdUS) but i don't feel like going in vc, would prob still be faster

**Comment by Coolykoen** [comment]
will do once we continue, thanks

**Comment by Coolykoen** [comment]
I think i'm just going to sell the tablet... All this just to use it is kind of not worth it haha.
Because even if we fix the current issue, there is still another issue that probably belongs in a new thread.
The pen only turns on if the oem driver is used, its a rechargable pen without a button... it just sleeps until the tablet detects it and then it turns on, but since that only works on the oem driver, i have to keep the pen on the charger for it to work with any other driver... :D

**Comment by Mrcubix** [comment]
there is prob a way to fix that too, just gotta find it

**Comment by jamesbt365** [comment]
Closing due to no activity.
